development.md

```diff
@@ -0,0 +1,156 @@
+because when forwarded it isn't available to your host.
+
+I had some success with QEmu on Linux for this, you need to use:
+
+```
+device_add usb-host,id=razer,hostbus=5,hostaddr=2
+```
+
+To add a device to qemu (change hostbus/hostaddr to the Razer device)
+
+and:
+
+```
+device_del razer
+```
+
+To remove the device from the virtual machine.
+
+If you're on MacOS or Windows I recommend to use VirtualBox and use USB passthough as described here:
+[VirtualBox USB Support](https://www.virtualbox.org/manual/ch03.html#settings-usb)
+
+Make sure you're using the same kernel version on both your development as your test machine. Easiest way
+to achieve this is to use the same Linux distribution and use the latest available kernel version.
+
+#### Debian/Ubuntu
+
+For Debian/Ubuntu install the following packages for kernel development:
+
+```
+apt-get install -y make gcc flex bison bc linux-headers-$(uname -r)
+```
+
+#### RedHat/Fedora
+
+For RedHat/Fedora install the following packages for kernel development:
+
+```
+dnf install -y binutils make gcc ncurses-devel sed flex bison kernel-headers
+```
+
+#### Build the kernel module
+
+Run the following command to build the kernel module:
+
+```
+make driver
+```
+
+This will build the kernel modules in `driver/*.ko`. If you make changes to the sources
+you can rerun the `make driver` command and the modules will be rebuild.
+
+
+#### Test the kernel module
+
+You can copy the `driver/*.ko` modules to your test system and (re)load for example
+the keyboard module using:
+
+```
+rmmod razerkbd ; insmod razerkbd.ko
+```
+
+If you update the kernel module you have to manually reload the module.
+
+Driver messages can be found by running:
+
+```
+dmesg
+```
+
+### Python daemon
+
+OpenRazer can only use your system site-packages and cannot be (easily) installed in
+a virtualenv.
+
+#### Debian/Ubuntu
+
+For Debian/Ubuntu install the following packages to be able to run the daemon:
+
+```
+apt-get install -y python3 python3-daemonize python3-dbus python3-gi python3-notify2 python3-numpy python3-pyudev python3-setproctitle
+```
+
+#### RedHat/Fedora
+
+For RedHat/Fedora install the following packages to be able to run the daemon:
+
+```
+dnf install -y python3 python3-daemonize python3-dbus python3-notify2 python3-numpy python3-pyudev python3-setproctitle
+```
+
+#### Test the daemon
+
+Copy or have another clone of openrazer on your test system.
+
+Run:
+```
+make install_i_know_what_i_am_doing
+```
+
+If you make changes to the Python sources this command has to be re-executed.
+
+It's recommended not to let the daemon fork to the backgrond so you can easily restart and see any error messages.
+This can be done by running:
+
+```
+/usr/bin/openrazer-daemon --foreground
+```
+
+## I want to contribute my changes!!1
+
+### Prerequisites
+
+The OpenRazer project is licensed using the CC-BY-SA-4.0 license and GPL-2.0-or-later licenses. Make sure you
```

Comment:
> CC-BY-SA-4.0 license

Iirc only the icon is licensed this way, the code is GPL

Reply:
makes sense, fixed in 39b5c3609cd90c80bda7fec13dcc2cb808db7068